dvc/data/add.py

```diff
@@ -0,0 +1,58 @@
+from typing import TYPE_CHECKING, Tuple, Union
+
+from .stage import stage
+from .transfer import transfer
+from .tree import Tree
+
+if TYPE_CHECKING:
+    from dvc.data.meta import Meta
+    from dvc.fs.base import FileSystem
+    from dvc.ignore import DvcIgnore
+    from dvc.objects.db import ObjectDB
+    from dvc.objects.file import HashFile
+    from dvc.types import AnyPath
+
+
+def add(
+    odb: "ObjectDB",
+    fs: "FileSystem",
+    fs_path: "AnyPath",
+    name: str,
```

Comment:
We shouldn't depend on fs.PARAM_CHECKSUM, because it will go away and doesn't really reflect what is going on here. E.g. hash type should be arbitrary and not depend on fs. E.g. think about sha* migration in 3.0, where it might be passed somewhere here as a name or might be a setting somewhere else, but definitely not in fs.